elyra/pipeline/pipeline.py

```diff
@@ -61,11 +61,11 @@ def __init__(self, id, type, classifier, filename, runtime_image, dependencies=N
         self._classifier = classifier
         self._filename = filename
         self._runtime_image = runtime_image
-        self._dependencies = dependencies or []
+        self._dependencies = Operation.scrub_list(dependencies)
         self._include_subdirectories = include_subdirectories
-        self._env_vars = env_vars or []
-        self._inputs = inputs or []
-        self._outputs = outputs or []
+        self._env_vars = Operation.scrub_list(env_vars)
+        self._inputs = Operation.scrub_list(inputs)
+        self._outputs = Operation.scrub_list(outputs)
         self._parent_operations = parent_operations or []
```

Comment:
Since `parent_operations` are collected differently and we already guarantee that no operations result in the empty list, we can remove the `or []` clause here.

Reply:
Going back to the assumption that these are optional parameters, I have restored the same implementation that we have in the master branch today,  where if not passed, a list parameter is initialized as an empty list